bokeh/document.py

```diff
@@ -752,10 +752,6 @@ def validate(self):
             None
 
         '''
-        # logging.basicConfig is a no-op if there's already
-        # some logging configured. We want to make sure warnings
-        # go somewhere so configure here if nobody has.
-        logging.basicConfig(level=logging.INFO)
```

Comment:
> all that matters is that these usage validation message are not silent by default under any circumstance

That is ensured by the fact that logging is configured at startup. Perhaps you can try to PR to check that it does work? I don't know how to exercise this.